controllers/users.js

```diff
@@ -0,0 +1,224 @@
+const Address = require("../models/address");
+const Images = require("../models/images");
+const resetToken = require("../models/resetToken");
+const Token = require("../models/access_token");
+const User = require("../models/User");
+const bcrypt = require("bcrypt");
+const cloudinary = require("cloudinary");
+const sendEmail = require("../utils/sendEmail");
+const jwt = require("jsonwebtoken");
+require("dotenv").config();
+
+cloudinary.config({
+  cloud_name: process.env.CLOUD_NAME,
+  api_key: process.env.API_KEY,
+  api_secret: process.env.API_SECRET,
+});
+
+const register = async (req, res) => {
+  try {
+    const user = new User();
+    const salt = await bcrypt.genSalt(10);
+    user.username = req.body.username;
+    user.email = req.body.email;
+    user.firstname = req.body.firstname;
+    user.lastname = req.body.lastname;
+    password = req.body.password;
+    con_password = req.body.con_password;
+    if (password == con_password) {
+      user.password = await bcrypt.hash(password, salt);
+      await sendEmail(
+        user.email,
+        "registration",
+        "user registered successfully"
+      );
+      await user.save();
+      res.send("successful");
+    }
+  } catch (err) {
+    res.send("failure");
+  }
+};
+
+const login = async (req, res) => {
+  const user = await User.findOne({ username: req.body.username });
+  try {
+    if (user) {
+      const validPassword = await bcrypt.compare(
+        req.body.password,
+        user.password
+      );
+      if (validPassword) {
+        let token = new Token();
+        let access_token = jwt.sign(
+          { userId: user._id },
+          process.env.SECRET_KEY,
+          {
+            expiresIn: "1h",
+          }
+        );
+        token.userId = user._id;
+        token.token = access_token;
+        let savedToken = await token.save();
+        res.send(savedToken);
+      }
+    }
+  } catch (err) {
+    res.send("failure");
+  }
+};
+
+const deleteUser = async (req, res) => {
+  try {
+    let user = req.user;
+    if (!user) {
+      res.send("user not found");
+    }
+    await user.remove();
```

Comment:
use user id to delete the user data,
and on delete the user data delete other things as well